common/parse.go

```diff
@@ -0,0 +1,74 @@
+package common
+
+import (
+	"strings"
+
+	"github.com/spyzhov/ajson"
+)
+
+// ParseResult parses the response from a provider into a ReadResult. A 2xx return type is assumed.
+// The sizeFunc, recordsFunc, nextPageFunc, and structureFunc are used to extract the relevant data from the response.
+// The sizeFunc returns the total number of records in the response.
+// The recordsFunc returns the records in the response.
+// The nextPageFunc returns the URL for the next page of results.
+// The structureFunc is used to structure the data into an array of ReadResultRows.
+// The fields are used to populate ReadResultRow.Fields.
+func ParseResult(
+	data *ajson.Node,
+	sizeFunc func(*ajson.Node) (int64, error),
+	recordsFunc func(*ajson.Node) ([]map[string]any, error),
+	nextPageFunc func(*ajson.Node) (string, error),
+	structureFunc func([]map[string]any, []string) ([]ReadResultRow, error),
```

Comment:
marshalFunc might be a better name, since you're essentially marshalling the result into ReadResultRows